v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/cassandra/mappings/CassandraMappings.java

```diff
@@ -108,6 +119,46 @@ private <T> void putSet(
                   CassandraRowValueArrayMapper.create(rowValueMapper)));
     }
 
-    public abstract CassandraMappings build();
+    abstract CassandraMappings autoBuild();
+
+    public CassandraMappings build() {
+      addMapBindings();
+      return autoBuild();
+    }
+
+    /** Add bindings for handling Cassandra `Map` Types to Json. */
+    private void addMapBindings() {
+      /* Add Mappings for Map Types.*/
+      ImmutableMap<String, Class> classMappings = typeToClassMappingBuilder().build();
+      ImmutableMap<String, CassandraFieldMapper<?>> fieldMappers = fieldMappingBuilder().build();
+      ImmutableMap<String, UnifiedTypeMapping> typeMappings = typeMappingBuilder().build();
+      ImmutableSet<String> primitiveTypes = classMappings.keySet();
+
+      for (String key : primitiveTypes) {
+        Class keyClass = classMappings.get(key);
+        for (String value : primitiveTypes) {
+          // The Map type as represented in Cassandra Schema
+          String mapType = "MAP<" + key.toUpperCase() + "," + value.toUpperCase() + ">";
```

Comment:
Does cassandra support all primitive types as key and values? like are decimal/timestamp/date types valid key types?

Reply:
Except for `Duration` all are valid. `Duration` is a newly added type.
Even for` Duration`  if there's a support in near future, our mapping is going to work, basically `getMap(class, class)` interface from `Cassandra` will be same . 
In our UT, we have all types covered as a key once and value once.